index.html

```diff
@@ -38,7 +40,7 @@ <h2>Add New Rate</h2>
             <label for="rate">Rate:</label>
             <input type="number" id="rate" name="rate" step="0.0001" required />
 
-            <button type="button" onclick="insertRate()">Insert Rate</button>
+            <button type="submit" onsubmit="insertRate()">Insert Rate</button>
```

Comment:
As I said in my previous comment, use the function `addEventListener` in the JS file to bind events to a DOM element.
Also, the submit event is triggered by the form, not the button. See: https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event

Reply:
@vdaguenet Thank you for your feedback, I have now implemented your recommendations

Reply:
Nice!